packages/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -119,33 +122,79 @@ export class ConnectionPropsForSessCfg {
             finalSessCfg.tokenValue = cmdArgs.tokenValue;
         }
 
+        // This function will provide all the needed properties in one array
+        if (optsToUse.getAllValuesBack) {
```

Comment:
What happens if both properties are set on the options object? Currently it looks like if getValueBack and getAllValuesBack are both set, then getAllValuesBack will take priority. Is this the expected behavior, or should we throw an error if both are set saying that the properties are mutually exclusive?

Reply:
I don't know. I mean, they can throw in 2 different functions in those options. What do you think @zFernand0 @crawr @katelynienaber ?

Reply:
I am not really sure why we need all of the following: `getValuesBack, getAllValuesBack, doPrompting`.

Couldn't we just ask that people pass in a prompting function if they want to prompt? And in the CLI, wherever `addPropsOrPrompt` was called with `doPrompting = true` before, we just pass in the prompting block that used to be inside of `addPropsOrPrompt` itself (the block starting at line [164](https://github.com/zowe/imperative/blob/5eb5c5b0ec8069bf7a1e119df6ee6222eff01807/packages/rest/src/session/ConnectionPropsForSessCfg.ts#L164) would be removed). Then we can tell whether they want to prompt by whether they passed in a function, and it would get rid of confusion surrounding `doPrompting`. If they passed in the function, we prompt. Otherwise, we return an incomplete session (or an array of all the fields that were missing).

If we really have to keep `doPrompting`, I don't see any difference between returning user input one field at a time, or all at once. I think giving people the option to decide how they want it is confusing. They can just receive all the prompted values in an array (maybe an array with just one value) and if they want to handle them one at a time they can do that in their code.

Reply:
@katelynienaber We do have to keep `doPrompting` for now, unless we're OK with a breaking change.

> They can just receive all the prompted values in an array (maybe an array with just one value) and if they want to handle them one at a time they can do that in their code.

That is a good point. I believe the discussion in #422 resulted in a decision to give people an option between returning user input one field at a time, or all at once. But perhaps this option causes unnecessary confusion? @jellypuno @zFernand0

Reply:
In my opinion, we could implement it like this for now then if users think that it is confusing then we can re-factor it. If I look at it 10 years for now, there might be features that can be added to getAllValuesBack that is not available in getValueBack.

Reply:
So.. I talked to @dkelosky  and he gave me his opinion about this matter. And since this 2 option is giving others confusion, I just decided to change it and follow Dan and Katelyn's suggestions. 

Current approach: There's one option called getValuesBack. Whether the external app wants to prompt the values one by one or all at once then it is the apps decision. doPrompting is set to false when getValuesBack is  not null. If doPrompting === true then it will trigger the original approach.

@tjohnsonBCM @katelynienaber @zFernand0 Please let me know what you think. I really want this PR to move forward because the are 2 PRs in Zowe Explorer that is dependent on this.